healthcare/api-client/v1/datasets/datasets.py

```diff
@@ -0,0 +1,425 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+from google.oauth2 import service_account
+
+
+# [START healthcare_get_client]
+def get_client():
+    """Returns an authorized API client by discovering the Healthcare API and
+    creating a service object using the service account credentials JSON."""
+    api_version = 'v1'
+    service_name = 'healthcare'
+
+    credentials = service_account.Credentials.from_service_account_file(
+        filename=os.environ["GOOGLE_APPLICATION_CREDENTIALS"],
+        scopes=["https://www.googleapis.com/auth/cloud-platform"],
+    )
+
+    return discovery.build(
+        service_name,
+        api_version,
+        credentials=credentials)
+# [END healthcare_get_client]
+
+
+# [START healthcare_create_dataset]
+def create_dataset(
+        project_id,
+        cloud_region,
+        dataset_id):
+    """Creates a dataset."""
+    client = get_client()
+    dataset_parent = 'projects/{}/locations/{}'.format(
+        project_id, cloud_region)
+
+    body = {}
+
+    request = client.projects().locations().datasets().create(
+        parent=dataset_parent, body=body, datasetId=dataset_id)
+
+    try:
+        response = request.execute()
+        print('Created dataset: {}'.format(dataset_id))
+        return response
+    except HttpError as e:
+        print('Error, dataset not created: {}'.format(e))
+        return ""
+# [END healthcare_create_dataset]
+
+
+# [START healthcare_delete_dataset]
+def delete_dataset(
+        project_id,
+        cloud_region,
+        dataset_id):
+    """Deletes a dataset."""
+    client = get_client()
+    dataset_name = 'projects/{}/locations/{}/datasets/{}'.format(
+        project_id, cloud_region, dataset_id)
+
+    request = client.projects().locations().datasets().delete(
+        name=dataset_name)
+
+    try:
+        response = request.execute()
+        print('Deleted dataset: {}'.format(dataset_id))
+        return response
+    except HttpError as e:
+        print('Error, dataset not deleted: {}'.format(e))
+        return ""
+# [END healthcare_delete_dataset]
+
+
+# [START healthcare_get_dataset]
+def get_dataset(
+        project_id,
+        cloud_region,
+        dataset_id):
+    """Gets any metadata associated with a dataset."""
+    client = get_client()
+    dataset_name = 'projects/{}/locations/{}/datasets/{}'.format(
+        project_id, cloud_region, dataset_id)
+
+    datasets = client.projects().locations().datasets()
+    dataset = datasets.get(name=dataset_name).execute()
+
+    print('Name: {}'.format(dataset.get('name')))
+    print('Time zone: {}'.format(dataset.get('timeZone')))
+
+    return dataset
+# [END healthcare_get_dataset]
+
+
+# [START healthcare_list_datasets]
+def list_datasets(project_id, cloud_region):
```

Comment:
`list` can potentially fail too, so why not checking the failure?

Reply:
IIRC there was not a consensus on whether error checking was supposed to be a part of GCP samples. If it's become the standard that we do check for errors, I can add that.

Reply:
OK I can live with it